### ISSUE TITLE
Features of the quiz website

### ISSUE THREAD

**Issue by Sanctogiacomo** [Sanctogiacomo/Quiz-Website#1]
FRONT-END TOOLS 
(HTML, CSS, fontawesome, bootstrap and jquery)
BACK-END TOOLS
(To be decided)
HOSTING
(To be decided)

PAGES/SECTIONS
An Homepage
A sign up section
A login section
A quiz page
A chat section/help section
An about us page

**Comment by Pythonian** [comment]
Let me get the conversation going.

To decide on what backend to be used, would this project strictly be web-based or is there a possibility of porting it into a mobile app?

Where to host would be determined by the backend tools to use. But in most cases, some hosting platforms like DigitalOcean or Heroku offer support to multiple languages.

For the Pages,

Homepage: What is an anonymous user expected to see and what would an authenticated user see different from what an anonymous user sees?

Definitely there should be a full-blown authentication functionality like Register / Login / Password Reset / Change password. But then does a user need to confirm his email before being able to log in? And what are the relevant data that would be required while a User registers?
Would a 3rd party OAuth service be used i.e Facebook Login / Twitter Login? or would they register using their school email or ???

Chat section? Like real-time chat with other users or chat with the staff/admin?

Then to the actual quiz functionalities, some questions:

1. Can anyone create a quiz or just the admins?
2. What type of quiz, multi-choice, true-or-false or essay type quiz?
3. Can anon users also take the quiz or only registered users?
4. Would the quiz be timed?
5. Can a User take a quiz just once or can take it multiple times?

That's that for now. Cheers!

**Comment by Sanctogiacomo** [comment]
I still don't want to limit the project to Web alone. It will be a nice thing if we can have a mobile app version but the question is are there people who can do the job?

**Comment by IzyPro** [comment]
There are... Kudirat for one showed interest for Kotlin.

**Comment by Pythonian** [comment]
> I still don't want to limit the project to Web alone. It will be a nice thing if we can have a mobile app version but the question is are there people who can do the job?

Well, I don't think you have to wait for people to show interest in building the mobile app version. What should be done is to develop the project with mobile app in mind, and once the web-based version is complete and gets some traction, maybe then you'd see a growing interest in people who would want to take a dig at it by developing the mobile version.

The most important thing is to get the project off ground. And at this stage, getting the web-based developers for me is more important than looking for mobile devs.

**Comment by Pythonian** [comment]
> There are... Kudirat for one showed interest for Kotlin.

And you don't use Kotlin right? That means if she starts with Kotlin and by the time you're free to join in, you won't be able to contribute because you're familiar with another framework!

**Comment by Sanctogiacomo** [comment]
As it is now only Kudirat is using kotlin for Android dev. So I think is a good idea to choose a particular framework/stack majority uses.

**Comment by Pythonian** [comment]
Well, two weeks or more can pass by in trying to decide/choose the stack to be used based on what the majority in the class uses. Besides, most of us are beginners. I think someone should just close eye and list out every language that could be used for this project, if not ...

**Comment by IzyPro** [comment]
That's true.... Maybe we'll just start with web but create the API like kelechi said with mobile in mind.

**Comment by Pythonian** [comment]
> That's true.... Maybe we'll just start with web but create the API like kelechi said with mobile in mind.

Yea, that's the way to go.

So who are those starting? Who's writing the first set of codes?

**Comment by IzyPro** [comment]
I think UI/UX and the API can be worked on at the same time. Before front-end. But that'll be till January right?

**Comment by Pythonian** [comment]
> I think UI/UX and the API can be worked on at the same time. Before front-end. But that'll be till January right?

Yea from what was discussed in the group chat, the project begins in January. But the devs involved should know themselves before January and plan the collaboration methods they'll use once they start.

But as it is now, it's only @Sanctogiacomo  that I know for certain is part of the project, but I don't even know the role he will be playing.

**Comment by Gloweeh** [comment]
Wow...this project is a very nice one. Thumbs up to the initiators. I would really love to be a part of this. Really don't know much but would love to learn from this community of pros.

**Comment by Pythonian** [comment]
> Wow...this project is a very nice one. Thumbs up to the initiators. I would really love to be a part of this. Really don't know much but would love to learn from this community of pros.

@Gloweeh Hi dear, which area do you know & what aspect do you want to learn so we can know where to help you out?

**Comment by Gloweeh** [comment]
@pythonian
Thanks a bunch. I am currently focusing on front end and would also love to learn python
